### ISSUE TITLE
Document type name for documents of type "Final Opinion" that are not true final opinions

### ISSUE THREAD

**Issue by vrajmohan** [18F/openFEC#2188]
Some AO documents, of type "Final Opinion", are not true final opinions. E.g., the document named 
"Supplemental Material from Daniels Cablevision, Inc."  in the AO https://fec-stage-proxy.18f.gov/data/legal/advisory-opinions/1998-17/. We should figure out how to classify them. @nickykrause, @noahmanger: have I captured this right?

**Comment by vrajmohan** [comment]
I discussed this with FEC and was told that we should only look at documents of type "Final Opinion" whose name also matches the pattern "AO year-serial" (e.g. "AO 2003-14").

**Comment by vrajmohan** [comment]
Unfortunately this strategy does not work for some cases - see https://beta.fec.gov/data/legal/advisory-opinions/2007-11/.

**Comment by vrajmohan** [comment]
After further discussion with FEC, true final opinions are those where the filename matches the regex pattern "\d\d\d\d-\d?". I will close this when this requirement is reflected in the code.

**Comment by vrajmohan** [comment]
Finding too many exceptions to this rule:
2007-09 (false negative)
1987-09 (false positive)

My recommendation is to abandon this path and try to parse all documents of category "Advisory Opinion".

Tagging @noahmanger.

**Comment by noahmanger** [comment]
Arg! So meaning that it would return any citation that shows up in any document within an AO?

**Comment by vrajmohan** [comment]
Sorry for causing confusion, I fixed my comment above. Advisory Opinions may contain several documents each of which belongs to a category. The categories are "Final Opinion", "Draft Documents", "Votes", etc. We were originally trying to figure out which of these documents should be parsed for citations and were told that we should parse "Final Opinions". However, we noticed that some of the documents that were of category "Final Opinion" were not true _Final Opinions_. We next tried to look at the file name and ensure that it matched "AO YYYY-NN" where YYYY is the year and NN is the serial number. But even with this we see exceptions. I am now suggesting that we stick with parsing documents of category "Final Opinion", even though some of these may not be true Final Opinions.
